### PR TITLE
Updated AzureMobileServicesClient definition

### PR DESCRIPTION
**Pull request by jsiegmund**
Added the Push interface with methods and made some small changes according to the current state of the library. 

Source documentation found at https://github.com/Azure/azure-mobile-apps-js-client/tree/master/sdk/src.

### CODE REVIEW

File: azure-mobile-services-client/AzureMobileServicesClient.d.ts [hunk 1]
```diff
@@ -7,19 +9,53 @@ declare namespace Microsoft.WindowsAzure {
 
     // MobileServiceClient object based on Microsoft Azure documentation: http://msdn.microsoft.com/en-us/library/windowsazure/jj554219.aspx
     interface MobileServiceClient {
+        new (applicationUrl: string): MobileServiceClient;
         new (applicationUrl: string, applicationKey: string): MobileServiceClient;
```
Comment (mhegazy) [inline azure-mobile-services-client/AzureMobileServicesClient.d.ts]: do not add a new overload, instead add `?`:

```ts
new (applicationUrl: string, applicationKey?: string): MobileServiceClient;
```

File: azure-mobile-services-client/AzureMobileServicesClient.d.ts [hunk 2]
```diff
@@ -7,19 +9,53 @@ declare namespace Microsoft.WindowsAzure {
...
+         *
+         * @param apiName The custom api to invoke.
+         * @param options Contains additional parameter information, valid values are:
+         *     body: The body of the HTTP request.
+         *     method: The HTTP method to use in the request, with the default being POST,
+         *     parameters: Any additional query string parameters, 
+         *     headers: HTTP request headers, specified as an object.
+         * @param callback Optional callback accepting (error, results) parameters.
+         */
+        invokeApi(apiName: string, options: InvokeApiOptions, callback: (error: any, results: any) => void): asyncPromise;
```
Comment (mhegazy) [inline azure-mobile-services-client/AzureMobileServicesClient.d.ts]: Please mark optional parameters as optional using `?`.

`invokeApi(apiName: string, options?: InvokeApiOptions, callback?: (error: any, results: any) => void): asyncPromise;v`

File: azure-mobile-services-client/AzureMobileServicesClient.d.ts [hunk 3]
```diff
@@ -7,19 +9,53 @@ declare namespace Microsoft.WindowsAzure {
...
+        /**
+         * Register a push channel with the Mobile Apps backend to start receiving notifications.
+         *
+         * @param platform The device platform being used - wns, gcm or apns.
+         * @param pushChannel The push channel identifier or URI.
+         * @param templates An object containing template definitions. Template objects should contain body, headers and tags properties.
+         * @param secondaryTiles An object containing template definitions to be used with secondary tiles when using WNS.
+         * @param callback Optional callback accepting (error, results) parameters.
+         */
+        register(platform: string, pushChannel: string, templates: any, secondaryTiles: any, callback: (error: any, results: any) => void): void;
```
Comment (mhegazy) [inline azure-mobile-services-client/AzureMobileServicesClient.d.ts]: same here for optional paramters

File: azure-mobile-services-client/AzureMobileServicesClient.d.ts [hunk 4]
```diff
@@ -7,19 +9,53 @@ declare namespace Microsoft.WindowsAzure {
...
+        /**
+         * Invokes the specified custom api and returns a response object.
+         *
+         * @param pushChannel The push channel identifier or URI.
+         * @param callback Optional callback accepting (error, results) parameters.
+         */
+        unregister(pushChannel: string, callback: (error: any, results: any) => void): void;
```
Comment (mhegazy) [inline azure-mobile-services-client/AzureMobileServicesClient.d.ts]: and here.

**Comment by mhegazy** [comment]
thanks @jsiegmund. can you take a look at the CI failing tests

**Comment by jsiegmund** [comment]
Done, needed to remove the two lines added by Typings

**Pull request merged**